### PR TITLE
feat: added '!' and 'not' negation operators

### PR DESCRIPTION
**Pull request by housed**
# Added '!' and 'not' Negation Operators

Added '!' and 'not' negation operators to the Litexa's language syntax.

## Description

Simply put, added '!' and 'not' negation operators to the Litexa's language syntax.

There's also some minor, random tweaks to the codebase in this PR.

## Motivation and Context

Want to provide developers with more flexibility in the conditional expressions that they build.

## Testing

Integration tests written at `.../tests/data/conditionals/`.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project
- [x] My change requires a change to the documentation
- [x] I have updated the documentation accordingly
- [x] I have read the **README** document
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

[issues]: https://github.com/alexa-games/litexa/issues
[license]: https://www.apache.org/licenses/LICENSE-2.0
[cla]: http://en.wikipedia.org/wiki/Contributor_License_Agreement

### CODE REVIEW

File: docs/reference/README.md [hunk 1]
```diff
@@ -1008,13 +1008,23 @@ This would match 'cat', 'caat', 'caaat', etc.
...
 
-Note: Unlike [say](#say) statements, reprompt statements
-are *not* accumulative; only the most recent reprompt statement
-will be included in the response.
+```coffeescript
+say "one"
+say "two"
+reprompt # would result in Alexa reprompting "one two three"
+say "three"
+```
```
Comment (joseifert) [inline docs/reference/README.md]: This was ultimately removed since it's relatively useless; would revert the changes for this PR.

File: packages/litexa/src/parser/litexa.pegjs [hunk 2]
```diff
@@ -797,13 +799,23 @@ SayStatement
...
-will be included in the response.
+Alternatively, writing a standalone `reprompt` will add the entire pending
+response's `say` speech to the reprompt.
+
+```coffeescript
+say "one"
+say "two"
+reprompt # would result in Alexa reprompting "one two three"
+say "three"
+```
```
Comment (joseifert) [inline packages/litexa/src/parser/litexa.pegjs]: Same as per above comment, this "echo" reprompt was ultimately removed, so wouldn't include these changes in the PR, to avoid confusion.

File: docs/reference/README.md (hunk 1 above)
Comment (ayrc) [reply docs/reference/README.md]: Looks like you might be addressing these in your say-reprompt syntax PR; I would merge that one first to make reverting this easier.

File: packages/litexa/src/parser/function.coffee [hunk 3]
```diff
@@ -54,6 +55,30 @@ class lib.Expression
...
+    val = @val
+    if @val.toLambda?
+      val = @val.toLambda(options)
+    op = operatorMap[@op]
+    if @skipParentheses
+      "#{op}#{val}"
+    else
+      "(#{op}#{val})"
+
+
```
Comment (ayrc) [inline packages/litexa/src/parser/function.coffee]: sorry, looks like this piece was part of a different commit. Can you add here:
```coffeescript
  toString: ->
    "#{@op}#{@val}"
```
